### PR TITLE
Automatically parse the error result field in `sendTransaction`

### LINKED ISSUE #128: Decode XDR structures wherever possible.
**Is your feature request related to a problem? Please describe.**
It's annoying to decode base64-encoded XDR that comes from the raw JSON responses because:

 (a) you don't necessarily know what the type is without referencing the documentation
 (b) you need to actually write the decoding code and error-check it

**Describe the solution you'd like**
This should be part of `Server`s request methods: they should wrap and decode the raw responses wherever possible:

- [x] simulateTransaction
- [x] getTransaction
- [ ] sendTransaction
- [x] getLedgerEntries
- [x] getLatestLedger
- [x] getEvents

**Additional context**
The work for this has been started in #127 and #129. 

**Please :+1: or :-1: as feedback for continuing this feature!**

### PR DESCRIPTION
**Pull request by Shaptic**
In particular,

```diff
-    errorResultXdr?: string;
+    errorResult?: xdr.TransactionResult;
```

The error result field, if present, will automatically be decoded.
 
Closes #128.

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -626,7 +630,13 @@ export class Server {
   public async sendTransaction(
     transaction: Transaction | FeeBumpTransaction,
   ): Promise<SorobanRpc.SendTransactionResponse> {
-    return await jsonrpc.post(
+    return this._sendTransaction(transaction).then(parseRawSendTransaction);
```
Comment (sreuland) [inline src/server.ts:633]: for consideration, a test to demonstrate this new expectation, like `it provides an encoded error result`  in https://github.com/stellar/js-soroban-client/blob/decode-send-transaction/test/unit/server/send_transaction_test.js#L38

**Review by sreuland — APPROVED** [review]
looks good, left one comment to consider including some test coverage.

File: src/server.ts (hunk 1 above)
Comment (Shaptic) [reply src/server.ts:633]: good call :+1:

Comment (Shaptic) [reply src/server.ts:633]: done in 930ef87!

**Pull request merged**